source/s3_client.c

```diff
@@ -243,6 +254,17 @@ struct aws_s3_client *aws_s3_client_new(
         return NULL;
     }
 
+    if (client_config->max_part_size != 0 && client_config->memory_limit_in_bytes != 0 &&
+        client_config->max_part_size >
+            (client_config->memory_limit_in_bytes / s_default_max_part_size_to_mem_lim_multiplier)) {
```

Comment:
trivial: use existing helper function?
```suggestion
            s_default_max_part_size_based_on_mem_limit(client_config->memory_limit_in_bytes)) {
```